tests/konflux-demo/konflux-demo.go

```diff
@@ -40,464 +39,378 @@ import (
+				Expect(fw.AsKubeAdmin.CommonController.Github.CreateRef(componentRepositoryName, appSpec.ComponentSpec.GitSourceDefaultBranchName, appSpec.ComponentSpec.GitSourceRevision, componentNewBaseBranch)).To(Succeed())
+			})
+
+			// Component are imported from gitUrl
+			It(fmt.Sprintf("creates component %s (private: %t) from git source %s", appSpec.ComponentSpec.Name, appSpec.ComponentSpec.Private, appSpec.ComponentSpec.GitSourceUrl), Label(devEnvTestLabel), func() {
+				componentObj := appservice.ComponentSpec{
+					ComponentName: componentName,
+					Application:   appSpec.ApplicationName,
+					Source: appservice.ComponentSource{
+						ComponentSourceUnion: appservice.ComponentSourceUnion{
+							GitSource: &appservice.GitSource{
+								URL:           appSpec.ComponentSpec.GitSourceUrl,
+								Revision:      gitRevision,
+								Context:       appSpec.ComponentSpec.GitSourceContext,
+								DockerfileURL: appSpec.ComponentSpec.DockerFilePath,
 							},
 						},
+					},
+				}
+
+				component, err = fw.AsKubeAdmin.HasController.CreateComponent(componentObj, namespace, "", secret.Name, appSpec.ApplicationName, false, utils.MergeMaps(constants.ComponentPaCRequestAnnotation, constants.DefaultDockerBuildPipelineBundle))
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+
+			When("Component is created", func() {
+
+				It("triggers creation of a PR in the sample repo", func() {
+
+					var prSHA string
+					Eventually(func() error {
+						prs, err := fw.AsKubeAdmin.CommonController.Github.ListPullRequests(componentRepositoryName)
+						Expect(err).ShouldNot(HaveOccurred())
+						for _, pr := range prs {
+							if pr.Head.GetRef() == pacBranchName {
+								prNumber = pr.GetNumber()
+								prSHA = pr.GetHead().GetSHA()
+								return nil
+							}
+						}
+						return fmt.Errorf("could not get the expected PaC branch name %s", pacBranchName)
+					}, pullRequestCreationTimeout, defaultPollingInterval).Should(Succeed(), fmt.Sprintf("timed out when waiting for init PaC PR (branch %q) to be created against the %q repo", pacBranchName, componentRepositoryName))
+
+					// We actually don't need the "on-pull-request" PipelineRun to complete, so we can delete it
+					Eventually(func() error {
+						pipelineRun, err = fw.AsKubeAdmin.HasController.GetComponentPipelineRun(component.GetName(), appSpec.ApplicationName, fw.UserNamespace, prSHA)
+						if err == nil {
+							Expect(fw.AsKubeAdmin.TektonController.DeletePipelineRun(pipelineRun.Name, pipelineRun.Namespace)).To(Succeed())
+							return nil
+						}
+						return err
+					}, pipelineRunStartedTimeout, constants.PipelineRunPollingInterval).Should(Succeed(), fmt.Sprintf("timed out when waiting for init PaC PipelineRun to be present in the user namespace %q for component %q with a label pointing to %q", fw.UserNamespace, component.GetName(), appSpec.ApplicationName))
+
+				})
+
+				It("component build status is set correctly", func() {
```

Comment:
```suggestion
				It("verifies component build status", func() {
```
just so it reads better in logs, in case of failure.